auth-tokens-filter/src/main/java/com/palantir/tokens/auth/http/BearerTokenLoggingFilter.java

```diff
@@ -69,6 +72,13 @@ private void clearMdc() {
         MDC.remove(TOKEN_ID_KEY);
     }
 
+    /**
+     * Based on the structure check from {@link UnverifiedJsonWebToken#of(BearerToken)}.
+     */
+    private boolean hasJwtStructure(String rawAuthHeader) {
+        return rawAuthHeader.split("\\.").length == 3;
```

Comment:
instead of copying the array every time, any chance we could just do:

    return rawAuthHeader.chars().filter(x -> '.').count() == 2;

?

Reply:
it might be worth adding a 'UnverifiedJsonWebToken.tryParse(String)' instead, btw - it avoids putting this logic into this class (which probably shouldn't care). It can do the same check and then delegate to 'of' and then return Optional.empty, which would let us avoid the catching Throwable here as well.

Reply:
Wouldn't in that case `tryParse` be another version of `.of` method but without the exception?